Easydict/Swift/Service/OpenAI/BaseOpenAIService.swift

```diff
@@ -0,0 +1,251 @@
+
+        openAI.chatsStream(query: query, url: url) { [weak self] res in
+            guard let self else { return }
+
+            switch res {
+            case let .success(chatResult):
+                if let content = chatResult.choices.first?.delta.content {
+                    resultText += content
+                    handleResult(queryType: queryType, resultText: resultText, error: nil, completion: completion)
+                }
+            case let .failure(error):
+                handleResult(queryType: queryType, resultText: nil, error: error, completion: completion)
+            }
+        } completion: { [weak self] error in
+            guard let self else { return }
+
+            if let error {
+                print("chatsStream error: \(String(describing: error))")
+                completion(result, error)
+            } else {
+                // If already has error, we do not need to update it.
+                if result.error == nil {
+                    // Since it is more difficult to accurately remove redundant quotes in streaming, we wait until the end of the request to remove the quotes.
+                    let nsText = resultText as NSString
+                    resultText = nsText.tryToRemoveQuotes()
+                    handleResult(queryType: queryType, resultText: resultText, error: nil, completion: completion)
+                }
+            }
+        }
+    }
+
+    // swiftlint:enable identifier_name
+
+    // MARK: Internal
+
+    var model: String {
+        get {
+            var model = Defaults[.openAIModel].rawValue
+            if model.isEmpty {
+                model = availableModels.first ?? OpenAIModel.gpt3_5_turbo_0125.rawValue
+            }
+            return model
+        }
+
+        set {
+            // easydict://writeKeyValue?EZOpenAIModelKey=gpt-3.5-turbo
+            let mode = OpenAIModel(rawValue: newValue) ?? .gpt3_5_turbo_0125
+            Defaults[.openAIModel] = mode
+        }
+    }
+
+    var apiKey: String {
+        // easydict://writeKeyValue?EZOpenAIAPIKey=
+
+        var apiKey = UserDefaults.standard.string(forKey: EZOpenAIAPIKey) ?? ""
+        if apiKey.isEmpty, Configuration.shared.beta {
+            apiKey = defaultAPIKey
+        }
+
+        return apiKey
+    }
+
+    var endPoint: String {
+        // easydict://writeKeyValue?EZOpenAIEndPointKey=
+
+        var endPoint = UserDefaults.standard.string(forKey: EZOpenAIEndPointKey) ?? ""
+        if endPoint.isEmpty {
+            endPoint = "https://\(host)/v1/chat/completions"
+        }
+
+        if !hasPrivateAPIKey() {
+            endPoint = defaultEndPoint
+        }
+
+        return endPoint
+    }
+
+    var availableModels: [String] {
+        OpenAIModel.allCases.map { $0.rawValue }
+    }
+
+    // MARK: Private
+
+    private var host: String {
+        // easydict://writeKeyValue?EZOpenAIDomainKey=
+        var host = UserDefaults.standard.string(forKey: "EZOpenAIDomainKey") ?? ""
```

Comment:
Maybe define a constant key and use `Defaults` to access it if we still need this.

Reply:
`EZOpenAIDomainKey` should be deprecated, I have removed it now.